### PR TITLE
add highly_variable_features

### PR DESCRIPTION
**Pull request by xinyuejohn**
**PR Checklist**

-   [ ] This comment contains a description of changes (with reason)
-   [ ] Referenced issue is linked
-   [ ] If you've fixed a bug or added code that should be tested, add tests!
-   [ ] Documentation in `docs` is updated

**Description of changes**

- Add highly_variable_features in ehrapy/preprocessing/_scanpy_pp_api.py.
	A normalized variance for each feature is computed. And features are ranked by the normalized variance.

**Technical details**

- additional dependencies: scikit-misc is added to poetry.toml because of the internal call of sc.pp.highly_variable_genes.
- added tests and updated docs

### CODE REVIEW

**Comment by Zethson** [comment]
Yeah, I guess this is the easy way for now.

Would it be possible for you to add like 1 test and expose this to the docs and import it in the tools init etc?

**Comment by xinyuejohn** [comment]
> Yeah, I guess this is the easy way for now.
> 
> Would it be possible for you to add like 1 test and expose this to the docs and import it in the tools init etc?

no problem

**Comment by xinyuejohn** [comment]
> Yeah, I guess this is the easy way for now.
> 
> Would it be possible for you to add like 1 test and expose this to the docs and import it in the tools init etc?

`from ehrapy.preprocessing._scanpy_pp_api import *  # noqa: E402,F403`
In the __init__.py it's like this, should I change this?

**Comment by Zethson** [comment]
> > Yeah, I guess this is the easy way for now.
> > Would it be possible for you to add like 1 test and expose this to the docs and import it in the tools init etc?
> 
> `from ehrapy.preprocessing._scanpy_pp_api import * # noqa: E402,F403` In the **init**.py it's like this, should I change this?

Ah whops. I'd move the new function into its own Python file. It is somewhat the Scanpy API, but I've decided to move all custom stuff elsewhere.

**Comment by xinyuejohn** [comment]
@Zethson Could you have a look at the failing windows test? Don't know what happened to scikit-misc package.

**Comment by Zethson** [comment]
It's this issue https://github.com/has2k1/scikit-misc/issues/4

**Comment by xinyuejohn** [comment]
> It's this issue [has2k1/scikit-misc#4](https://github.com/has2k1/scikit-misc/issues/4)

great! thanks!

**Comment by xinyuejohn** [comment]
> It's this issue [has2k1/scikit-misc#4](https://github.com/has2k1/scikit-misc/issues/4)

do we need to install [numpy+mkl](https://www.lfd.uci.edu/~gohlke/pythonlibs/#numpy) and then install from a .whl from [lfd.uci.edu/~gohlke/pythonlibs/#scikit-misc](https://www.lfd.uci.edu/~gohlke/pythonlibs/#scikit-misc)?

Not sure how to handle this here

**Comment by xinyuejohn** [comment]
> You can install numpy from conda (it comes with mkl) and the skmisc wheel from [lfd.uci.edu/~gohlke/pythonlibs/#scikit-misc](https://www.lfd.uci.edu/~gohlke/pythonlibs/#scikit-misc)

thanks! I would try again! @Koncopd

File: pyproject.toml [hunk 1]
```diff
@@ -43,6 +43,7 @@ medcat = {version = ">=1.2.6", optional = true}
 anndata = ">=0.7.8"
 fancyimpute = "^0.7.0"
```
Comment (Zethson) [inline pyproject.toml:44]: Could you please for all of them use >= and not ^?

File: tests/preprocessing/test_highly_variable_features.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import sys
+
+import pytest
+
+import ehrapy as ep
+from ehrapy.preprocessing.highly_variable_features import highly_variable_features
+
+if sys.platform.startswith("win"):
```
Comment (Zethson) [inline tests/preprocessing/test_highly_variable_features.py:8]: This is no longer required, is it?

Comment (Zethson) [reply tests/preprocessing/test_highly_variable_features.py:8]: We could also keep it for developers who are on Windows.... But I think those should properly setup an environment anyways

File: pyproject.toml (hunk 1 above)
Comment (xinyuejohn) [reply pyproject.toml:44]: no problem

**Pull request merged**